libs/langchain/langchain/chat_models/vertexai.py

```diff
@@ -256,6 +276,27 @@ def _stream(
                 run_manager.on_llm_new_token(response.text)
             yield ChatGenerationChunk(message=AIMessageChunk(content=response.text))
 
+    async def _astream(
+        self,
+        messages: List[BaseMessage],
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[AsyncCallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> AsyncIterator[ChatGenerationChunk]:
+        question = _get_question(messages)
```

Comment:
nits: the next few lines seem to be a duplication

It might be a good idea to create an internal method `_start_chat` and reuse it for both streaming and not streaming implementations, wdyt?

Reply:
sure. Will do it and rebase with master

Reply:
@lkuligin it should be good now. I moved the logic to the existing start_chat. Tests are all passing.